proxy/hdrs/URL.cc

```diff
@@ -1765,8 +1765,10 @@ url_MD5_get_general(const URLImpl *url, CryptoContext &ctx, CryptoHash &hash, ca
   ctx.update(&port, sizeof(port));
   if (generation != -1) {
     ctx.update(&generation, sizeof(generation));
+    Debug("url", "Final url string for hashing %.*s%d%d", p - buffer, buffer, port, generation);
```

Comment:
For better readability, I would do the p - buffer calculation once (above the conditional p != buffer) and assign it to a well named const variable.